apps/mobile/src/components/OfferDetailScreen/components/OfferInfo.tsx

```diff
@@ -43,13 +43,15 @@ function OfferInfo({offer}: {offer: OneOfferInState}): JSX.Element {
   return (
     <Stack f={1} mx={'$2'} my={'$4'}>
       <ScreenTitle text={t('offer.title')}>
-        <IconButton
-          variant={'dark'}
-          icon={flagSvg}
-          onPress={() => {
-            void reportOffer(offer.offerInfo.offerId)()
-          }}
-        />
+        {!offer.flags.reported && (
```

Comment:
I understand this is a good practice to do. But when can offer be reported and at the same time displayed? That should not happen right?

Reply:
This can happen when user does like so:
Request offer -> Go to detail -> Go to chat -> report offer -> after reporting it goes back to offer detail -> and there should not be the flag option available (maybe this is sth to discuss if we should not redirect in all cases to marketplace)